internal/services/cognitive/cognitive_account_resource.go

```diff
@@ -687,76 +697,76 @@ func expandCognitiveAccountNetworkAcls(d *pluginsdk.ResourceData) (*cognitiveser
 			value := v.(map[string]interface{})
 			subnetId := value["subnet_id"].(string)
 			subnetIds = append(subnetIds, subnetId)
-			rule := cognitiveservices.VirtualNetworkRule{
-				ID:                               utils.String(subnetId),
+			rule := cognitiveservicesaccounts.VirtualNetworkRule{
+				Id:                               subnetId,
 				IgnoreMissingVnetServiceEndpoint: utils.Bool(value["ignore_missing_vnet_service_endpoint"].(bool)),
 			}
 			networkRules = append(networkRules, rule)
 		}
 	}
 
-	ruleSet := cognitiveservices.NetworkRuleSet{
-		DefaultAction:       cognitiveservices.NetworkRuleAction(defaultAction),
-		IPRules:             &ipRules,
+	ruleSet := cognitiveservicesaccounts.NetworkRuleSet{
+		DefaultAction:       &defaultAction,
+		IpRules:             &ipRules,
 		VirtualNetworkRules: &networkRules,
 	}
 	return &ruleSet, subnetIds
 }
 
-func expandCognitiveAccountStorage(input []interface{}) *[]cognitiveservices.UserOwnedStorage {
+func expandCognitiveAccountStorage(input []interface{}) *[]cognitiveservicesaccounts.UserOwnedStorage {
 	if len(input) == 0 {
 		return nil
 	}
-	results := make([]cognitiveservices.UserOwnedStorage, 0)
+	results := make([]cognitiveservicesaccounts.UserOwnedStorage, 0)
 	for _, v := range input {
 		value := v.(map[string]interface{})
-		results = append(results, cognitiveservices.UserOwnedStorage{
-			ResourceID:       utils.String(value["storage_account_id"].(string)),
-			IdentityClientID: utils.String(value["identity_client_id"].(string)),
+		results = append(results, cognitiveservicesaccounts.UserOwnedStorage{
+			ResourceId:       utils.String(value["storage_account_id"].(string)),
+			IdentityClientId: utils.String(value["identity_client_id"].(string)),
 		})
 	}
 	return &results
 }
 
-func expandCognitiveAccountIdentity(vs []interface{}) (*cognitiveservices.Identity, error) {
+func expandCognitiveAccountIdentity(vs []interface{}) (*identity.SystemUserAssignedIdentityMap, error) {
 	if len(vs) == 0 {
-		return &cognitiveservices.Identity{
-			Type: cognitiveservices.ResourceIdentityTypeNone,
+		return &identity.SystemUserAssignedIdentityMap{
+			Type: "None",
```

Comment:
can we use the constants for this from `identity`?

Reply:
We can't as they're local to the `identity` package. Should we make them available outside the `identity` package?